cmd/sedad/utils/merkle.go

```diff
@@ -0,0 +1,107 @@
+package utils
+
+import (
+	"bytes"
+	"hash"
+	"math/bits"
+
+	"golang.org/x/crypto/sha3"
+)
+
+// RootFromEntries computes the root of the Merkle Hash Tree whose
+// leaves are the entries given in byte slices in the provided order.
+// It largely follows RFC-6962 with some modifications:
+//   - It uses SHA3-256 hashing algorithm.
+//   - Leaves are double-hashed for second preimage resistance. No hashing
```

Comment:
Same question here?

Reply:
We do still double hash the leaves if that is what you're referring to?

Reply:
Yeah since when we discussed it I thought we concluded that the preimage attack vector didn’t really apply to us + DR result IDs are already a hash of the dr result. No opinion, just curious to what we landed on ultimately

Reply:
Yes makes sense. I don't think we want to finalize the specs before we try out proving from the Solidity contract side, but for now:

- Leaves are sorted
- Each hash pair is sorted to make proofs more succinct.
- No hashing prefixes or double hashing leaves